### ISSUE TITLE
Do not write to superglobals

### ISSUE THREAD

**Issue by emanuele45** [elkarte/Elkarte#1655]
We did already improve the situation a tiny bit for 1.0, though there are still so many places where we write to superglobals...
Let's see how many.
Using this regex:

```
\$_(POST|REQUEST|GET)\[('[a-zA-Z0-9\-_]*'|('[a-zA-Z0-9\-_]*' \. )*\$[a-zA-Z0-9\-_]*( \. '[a-zA-Z0-9\-_]*')*)\]\s*[\+\-\/\*]*={1}(>)*\s
```

I found 470+ occurrences. :fearful: :scream_cat: 
I was going to propose this to be a 1.1 blocker, but considering that number it's probably better wait until we have done some work on it before say anything...

**Comment by Spuds** [comment]
470+ => 241 now, and some of those are not ours :dancer:

**Comment by joshuaadickerson** [comment]
Stuff like ```$this->_req->query->tid = (int) $this->_req->query->tid;``` is still writing to a superglobal :(

**Comment by emanuele45** [comment]
It's not that the code is always straightforward enough to just scrap all the writings. ;)

**Comment by Spuds** [comment]
Yup, lots to do, lots to do,  and technically I believe if you were to go look at $_GET['tid'] its not an int after that statement runs?

**Comment by Spuds** [comment]
last I ran the regex its down to 71 in the sources ... some of those are just reads during init ...
